### PR TITLE
add configurable context filter to demo

### PR DESCRIPTION
**Pull request by sfc-gh-jreini**
Add configurable context filter to the snowflake demo.

- [x] Dependent on: #1192

### CODE REVIEW

File: trulens_eval/examples/snowflake_demo/common_ui.py [hunk 1]
```diff
@@ -251,6 +254,17 @@ def configure_model(
                 if model_config.use_rag != st.session_state[USE_RAG_KEY]:
                     st.session_state[USE_RAG_KEY] = model_config.use_rag
 
+                if model_config.retrieval_filter != st.session_state[RETRIEVAL_FILTER_KEY]:
```
Comment (sfc-gh-chu) [inline trulens_eval/examples/snowflake_demo/common_ui.py:257]: Shouldn't we set the state after `model_config.retrieval_filter` has been assigned by the slider?

**Review by sfc-gh-chu — CHANGES_REQUESTED** [review]

File: trulens_eval/examples/snowflake_demo/llm.py [hunk 2]
```diff
@@ -161,9 +164,13 @@ def retrieve_context(
         self, last_user_message: str, prompt_str: str,
         conversation: Conversation
     ):
-        texts = self.retriever.retrieve(query=last_user_message)
-        context_message = "\n\n".join(texts)
-        return _reencode_outputs(context_message), texts
+        @context_filter(f_small_local_models_context_relevance, conversation.model_config.retrieval_filter)
```
[the remark this answers is not in the document]
Comment (sfc-gh-jreini) [reply trulens_eval/examples/snowflake_demo/llm.py:167]: This is using the custom python implementation here (https://github.com/truera/trulens/pull/1192). Not quite the same as `WithFeedbackFilterDocuments`.

What do you mean add the guardrailing changes to `snowflake_demo` first? Isn't that what this change is doing?

File: trulens_eval/examples/snowflake_demo/common_ui.py (hunk 1 above)
Comment (sfc-gh-jreini) [reply trulens_eval/examples/snowflake_demo/common_ui.py]: Fixed!

File: trulens_eval/examples/snowflake_demo/common_ui.py [hunk 3]
```diff
@@ -133,10 +133,10 @@ def login():
 
 def get_tru_app_id(
     model: str, temperature: float, top_p: float, max_new_tokens: int,
-    use_rag: bool
+    use_rag: bool, retrieval_filter: float
 ):
     # Args are hashed for cache lookup
-    return f"app-prod-{model}{'-rag' if use_rag else ''} (temp-{temperature}-topp-{top_p}-maxtokens-{max_new_tokens})"
+    return f"app-prod-{model}{'-rag' if use_rag else ''}-retrieval-filter-{retrieval_filter} (temp-{temperature}-topp-{top_p}-maxtokens-{max_new_tokens})"
```
Comment (sfc-gh-dkurokawa) [inline trulens_eval/examples/snowflake_demo/common_ui.py:139]: if you don't use a RAG the retriever filter shouldn't matter no?

Comment (sfc-gh-jreini) [reply trulens_eval/examples/snowflake_demo/common_ui.py]: good point - updated

**Review by sfc-gh-chu — APPROVED** [review]

**Pull request merged**